### ISSUE TITLE
Spin wobble on FontAwesome Pro using Kit

### ISSUE THREAD

**Issue by mwseifert** [FortAwesome/Font-Awesome#16495]
**Describe the bug**
@tagliala has asked that I open this as a bug ticket, so that it can be investigated. There appears to be a wobble issue with regard to how FontAwesome Pro kits load partially, to keep the file size down.

**To Reproduce**
FontAwesome free: https://jsfiddle.net/j9edumwa/
FontAwesome Pro: https://jsfiddle.net/5uks64em/

Both using Web Fonts. Free is steady, and pro wobbles.

**Comment by tagliala** [comment]
@robmadole this only happens with Pro Kits CSS, it could be related to incremental font files

It is not reproducible with pro cdn: https://jsfiddle.net/tagliala/2trnyu4v/2/

**Comment by robmadole** [comment]
Yep, @tagliala is correct. It's related to the–we call them chunks–individual font files.

I have a fix in mind for this so I'll get this up on our work board.

**Comment by BladeMF** [comment]
I have the same issue.

**Comment by TarikFojnica** [comment]
Do we have any news on this?

**Comment by tagliala** [comment]
This has also been reported in #17469

@robmadole is there any chance to add an option to opt-out from the optimization causing this issue?

**Comment by Psychokiller1888** [comment]
This is still the case, using FA pro beta 6 kit

**Comment by hoRacy** [comment]
@robmadole 
It's very sad that the FA icons are **UNUSABLE** with kits due to lack of icon centering when spinning ;(
Is the PRO user worse than the normal?

**Comment by jirkadev** [comment]
@robmadole @tagliala we recently bought the pro version. And we see the same problem. If we switch to pro icons in the kit settings. Pseudo element :before is not "vertically centered" Could you reproduce this issue? I think this should be fixed!

For us, this issue is present in WebKit browsers (Chrome, Safari, Opera). This issue is not present in Firefox.

free icons (webfont)
<img width="452" alt="Screen Shot 2022-04-26 at 21 18 38" src="https://user-images.githubusercontent.com/2588479/165376221-ac6f742e-da3e-4881-8954-988f107ba4a3.png">

pro icons (webfont)
<img width="420" alt="Screen Shot 2022-04-26 at 21 17 49" src="https://user-images.githubusercontent.com/2588479/165376226-b44d87bb-29e1-4458-a414-f5a20ed21a7e.png">

**Comment by mlwilkerson** [comment]
We're taking another look at this. I have previously confirmed and reproduced it. What I have seen is that when `unicode-ranges` are used in a `@font-face` rule, the wobble problem occurs. A similar situation (same font, same glyphs) without a `unicode-range` does not have the wobble.

We use `unicode-range` in our Pro Webfont/CSS and not Free. The reason is that we have some additional optimization in Pro Webfont/CSS whereby the font set is chunked in subsets, allowing us to load only the webfont chunks your web pages actually use. The `unicode-range` is the mechanism for selecting which webfont chunk should be loaded for a given unicode.

It has been a mystery as to why some browsers lay out the glyphs differently when there is a `unicode-range` versus when there is not.

**Comment by mlwilkerson** [comment]
We've recently learned that some browsers determine the line height for their layout algorithms based on the metrics of the `space` character in the font (unicode `0x20`). Because our `unicode-range` values in the CSS for these webfont chunks don't specify `U+20` as one of the characters in the range, these browsers seem to be falling back to whatever font they next find that maps a space character, and they use the font metrics associated with that _other font_ instead. This is what has been causing the wobble in some browsers.

We've verified a fix that corrects the issue in Chrome, Safari, and Opera. I expect the fix to be included in a Font Awesome release soon.

Thanks to [Roel](https://pixelambacht.nl/) for tracking down the root cause!

**Comment by serkanyazicioglu** [comment]
Any update about this? Thanks...

**Comment by robmadole** [comment]
We've got a released planned in the next two weeks.

**Comment by zagarskas** [comment]
I've been using the following to "reduce" the wobble...
Seems only "certain magic numbers" work. lol
- Set `font-size: 3.3333em;` (which is 53.3328px in my case)
- Set `font-size: 5em;` (which is 80px in my case)
-  `line-height: 1;` add on `fw fa-spin-pulse`
-  worked well with `fa-duotone`
- results very as per icon used
- tried **_lotsastuff_** im just not going to list here. lol. prob could have switched to the JS kit by now. lol

**I figured the wobble has something to do with "uneven numbers" or irrational numbers?**
Maybe has something to do with PI and being able to find an "absolute center"?

Anyway... been waiting a long time for this. lol
so, just want to slide that 2015 date in there as a pro user. 99.3333% satisfied. (seriously, great product. use it eryday)
![image](https://user-images.githubusercontent.com/859427/187095935-a2182c27-939b-494d-8eae-2dbc24f3ff9b.png)

**Comment by No1e** [comment]
Hey guys,

what is the status of this issue? Is there a fix?

I am using 6.3.0 Pro version, and the issue is still there.

Best regards,
Nole

**Comment by tagliala** [comment]
Hi @michaelurban 

This specific issue was related to an improper subsetting of the Pro webfont, and it is now fixed.

> You guys really need to tell people about this issue so they don't throw their time down a bottomless pit.

A generic problem is related to the webfont implementation itself, we can't fix it, and it is documented at https://fontawesome.com/docs/web/style/animate#icon-animation-wobbles

Another thing which helps but is not documented is to use multiples of the base font size (16px) 

Ref: https://jsfiddle.net/tagliala/8djt9nur/10/


https://github.com/FortAwesome/Font-Awesome/assets/556268/0688aeb2-ed57-4f23-9e5c-3064cb20023b



If you can provide a reproducible test case, I will check if there are still issues with subsetting

Otherwise, if your application needs pixel perfect animations, please:
- Switch to the SVG implementation
- Use an SVG spinner for this specific use case

Hope it helps

**Comment by michaelurban** [comment]
@tagliala You're correct. it is an issue across the board with webfonts. Back to SVGs. Apologies.
